src/plugins/data/common/search/search_source/create_search_source.ts

```diff
@@ -33,18 +34,47 @@ export const createSearchSource = (
   indexPatterns: DataViewsContract,
   searchSourceDependencies: SearchSourceDependencies
 ) => {
-  const createFields = async (searchSourceFields: SerializedSearchSourceFields = {}) => {
+  let dataViewLazy: DataViewLazy | undefined;
+  const createFields = async (
+    searchSourceFields: SerializedSearchSourceFields = {},
+    useDataViewLazy = false
+  ) => {
     const { index, parent, ...restOfFields } = searchSourceFields;
     const fields: SearchSourceFields = {
       ...restOfFields,
     };
 
     // hydrating index pattern
     if (searchSourceFields.index) {
-      if (typeof searchSourceFields.index === 'string') {
-        fields.index = await indexPatterns.get(searchSourceFields.index);
+      if (!useDataViewLazy) {
+        if (typeof searchSourceFields.index === 'string') {
+          fields.index = await indexPatterns.get(searchSourceFields.index);
+        } else {
+          fields.index = await indexPatterns.create(searchSourceFields.index);
+        }
       } else {
-        fields.index = await indexPatterns.create(searchSourceFields.index);
+        if (typeof searchSourceFields.index === 'string') {
+          dataViewLazy = await indexPatterns.getDataViewLazy(searchSourceFields.index);
+          const dataView = new DataView({
+            spec: await dataViewLazy.toSpec(),
+            // field format functionality is not used within search source
+            fieldFormats: {} as FieldFormatsStartCommon,
+            shortDotsEnable: await searchSourceDependencies.dataViews.getShortDotsEnable(),
+            metaFields: await searchSourceDependencies.dataViews.getMetaFields(),
+          });
+
+          fields.index = dataView;
+        } else {
+          dataViewLazy = await indexPatterns.createDataViewLazy(searchSourceFields.index);
+          const dataView = new DataView({
+            spec: await dataViewLazy.toSpec(),
+            // field format functionality is not used within search source
+            fieldFormats: {} as FieldFormatsStartCommon,
+            shortDotsEnable: await searchSourceDependencies.dataViews.getShortDotsEnable(),
+            metaFields: await searchSourceDependencies.dataViews.getMetaFields(),
+          });
+          fields.index = dataView;
```

Comment:
> Do we even need DataViewLazy here?

I really don't see thew upside to having DataView objects with field lists of uncertain content.

Reply:
>And the uncertain content, you mean, that could be an already cached field list (wouldn't apply to the alerting rule usage, but elsewhere), correct?

Yes, correct. While this is fine for isolated circumstances it could cause real problems elsewhere. Even having a DataView with a partial field list and no explanation of what it contains could be confusing if you're debugging code thats far away from where the data view is populated.